tests/test.js

```diff
@@ -438,4 +438,173 @@ test('NbsDecoder.getPackets() returns the last packet of each type when given a
+  { type: pingType, subtype: 0 },
+  { type: pangType, subtype: 100 },
+  { type: pangType, subtype: 200 },
+];
+
+test('NbsDecoder.nextTimestamp() outputs the exact timestamp as input, given steps = 0', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  // const new_timestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 0);
+  const new_timestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 0);
+
+  assert.equal(
+    tsToBigInt(new_timestamp),
+    tsToBigInt(start),
+    'With steps = 0, input start timestamp should be equal to the returned timestamp for single type'
+  );
+});
+
+test('NbsDecoder.nextTimestamp() Stepping forward by one timestamp returns a timestamp which is greater than input.', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 1);
+  const multiTypeTimestamp = decoder.nextTimestamp(start, multiTypeNextTimestampArray, 1);
+
+  assert.ok(
+    tsToBigInt(singleTypeTimestamp) > tsToBigInt(start),
+    'The new return Nbs timestamp is greater than the input timestamp'
+  );
+
+  assert.ok(
+    tsToBigInt(multiTypeTimestamp) > tsToBigInt(start),
+    'The new return Nbs timestamp is greater than the input timestamp'
+  );
+});
+
+test(`NbsDecoder.nextTimestamp() Stepping forward by five returns the expected timestamp for single and multi type`, () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 5);
+  const multiTypeTimestamp = decoder.nextTimestamp(start, multiTypeNextTimestampArray, 5);
+
+  assert.equal(tsToBigInt(singleTypeTimestamp), tsToBigInt({ seconds: 1015, nanos: 0 }));
+  assert.equal(tsToBigInt(multiTypeTimestamp), tsToBigInt({ seconds: 1013, nanos: 0 }));
+});
+
+test('NbsDecoder.nextTimestamp() Stepping backward returns a timestamp which is less than input', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(end, { type: pingType, subtype: 0 }, -1);
+
+  const multiTypeTimestamp = decoder.nextTimestamp(end, multiTypeNextTimestampArray, -1);
+
+  assert.ok(
+    tsToBigInt(singleTypeTimestamp) < tsToBigInt(end),
+    'The return Nbs timestamp is less than the input timestamp for a single type'
+  );
+
+  assert.equal(tsToBigInt(singleTypeTimestamp), tsToBigInt({ seconds: 1894, nanos: 0 }));
+  ('The return Nbs timestamp exactly matches the expected timestamp');
+
+  assert.ok(
+    tsToBigInt(multiTypeTimestamp) < tsToBigInt(end),
+    'The return Nbs timestamp is less than the input timestamp for multiple types'
+  );
+});
+
+test('NbsDecoder.nextTimestamp() Stepping backwards/forwards respective of begin/end iterators return the correct timestamps.', () => {
```

Comment:
From the point of view of JS there are no iterators so I wouldn't mention them.
Also from the test name it's ambiguous what you're testing.
I'd put something more like "stepping past the end of the end iterator returns the final timestamp"
and split it into two tests.

Splitting these tests isn't expensive and it makes it clearer what each test is testing.

Reply:
Split up and renamed.